lib/index_documents.rb

```diff
@@ -0,0 +1,65 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    links = message.payload["links"].try(:dup)
+
+    if links
+      links.each do |(k, vs)|
+        paths = vs.map { |v| base_path(v) }.compact
```

Comment:
It would be good to sort the output for this, since the ordering we receive them in is not significant.  Sorting will ensure we get a consistent order out when we process the same document, and that in turn will ensure we don't send unnecessary changes to elasticsearch, allowing it to avoid doing work when there's no semantic change to the document.